pkg/skaffold/deploy/helm.go

```diff
@@ -76,6 +80,13 @@ func (h *HelmDeployer) Labels() map[string]string {
 func (h *HelmDeployer) Deploy(ctx context.Context, out io.Writer, builds []build.Artifact, labellers []Labeller) *Result {
 	event.DeployInProgress()
 
+	hv, err := h.binVer(ctx)
```

Comment:
Consider moving this to the constructor`NewHelmDeployer` to warn or fail fast in future when skaffold runner is initializing run config `runner.NewForConfig`

I feel we should move this to `debug` for now. Right now, users might not understand why the warning message is important. I am assuming in future, we would fallback to helm 2 or helm 3  and then warning might make more sense.
e.g.
```
failed to parse binary version: 3rc1. Assuming helm-2.X deployer.
```

Reply:
I had considered this, but constructors should not generate I/O events, as there could be extremely adverse performance impacts when doing so.